### ISSUE TITLE
[Bug]: jsx-curly-newline and comments

### ISSUE THREAD

**Issue by josemigallas** [jsx-eslint/eslint-plugin-react#3584]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

`jsx-curly-newline` reports an error when using multiline comments in JSX.

```jsx
      {/* <FormGroup
        helperTextInvalid={helperText}
        inputProps={inputProps}
        type="password"
      /> */}
```

```
  134:7   error  Unexpected newline after '{'                             react/jsx-curly-newline
  138:12  error  Unexpected newline before '}'                            react/jsx-curly-newline
  ```

`$ eslint --ext .js,.jsx,.ts,.tsx .`

### Expected Behavior

I would expect no error.

### eslint-plugin-react version

7.32.2

### eslint version

7.32.0

### node version

14.21.3

**Comment by ljharb** [comment]
if you removed the comment delimiters, what would happen?

**Comment by ROSSROSALES** [comment]
I am unable to recreate this error with your example. @josemigallas, were you able to resolve this issue?

**Comment by ljharb** [comment]
Closing, but will reopen if it can be reproduced.

**Comment by josemigallas** [comment]
Apologies, I lost track of this issue but I have not been able to fix it yet.

> if you removed the comment delimiters, what would happen?

Like this?

```jsx
      { <FormGroup
        helperTextInvalid={helperText}
        inputProps={inputProps}
        type="password"
      /> }
```

The error doesn't show up, but others do:
```
  57:7   error  Curly braces are unnecessary here                                             react/jsx-curly-brace-presence
  57:10  error  'FormGroup' is not defined                                                    react/jsx-no-undef
  58:28  error  Unsafe assignment of an `any` value                                           @typescript-eslint/no-unsafe-assignment
  59:21  error  Unsafe assignment of an `any` value                                           @typescript-eslint/no-unsafe-assignment
  61:7   error  The closing bracket must be aligned with the opening tag (expected column 9)  react/jsx-closing-bracket-location
  ```


It also doesn't show with one line comments like:
```jsx
{/* <FormGroup ... */}
```

**Comment by ROSSROSALES** [comment]
@josemigallas thanks for responding!
Just to make sure, is it possible to see your jsx-curly-newline rule options?

**Comment by josemigallas** [comment]
> @josemigallas thanks for responding! Just to make sure, is it possible to see your jsx-curly-newline rule options?

```
"react/jsx-curly-newline": ["error", { "multiline": "forbid", "singleline": "forbid" }],
```
